packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/SerializationTests.kt

```diff
@@ -0,0 +1,446 @@
+                expected as RealmAny?
+                actual as RealmAny?
+
+                if (expected != null && actual != null) {
+                    when (expected.type) {
+                        RealmAny.Type.OBJECT -> {
+                            assertEquals(expected.type, actual.type)
+                            assertEquals(
+                                expected.asRealmObject<SerializableSample>().stringField,
+                                actual.asRealmObject<SerializableSample>().stringField
+                            )
+                        }
+                        else -> assertEquals(expected, actual)
+                    }
+                } else if (expected != null || actual != null) {
+                    fail("One of the RealmAny values is null, expected = $expected, actual = $actual")
+                }
+            }
+            else -> assertEquals(expected, actual)
+        }
+    }
+
+    @Test
+    fun exhaustiveRealmTypesTester() {
+        val expected = SerializableSample().apply {
+            nullableObject = SerializableSample()
+            realmEmbeddedObject = SerializableEmbeddedObject()
+        }
+        val encoded: String = json.encodeToString(expected)
+        val decoded: SerializableSample = json.decodeFromString(encoded)
+
+        TypeDescriptor.elementTypesForSerializers
+            .map { it.classifier }
+            .filterNot { it == RealmAny::class } // tested in exhaustiveRealmAnyTester
+            .forEach { classifier ->
+                when (classifier) {
+                    Decimal128::class -> assertEquals(
+                        expected.decimal128Field,
+                        decoded.decimal128Field
+                    )
+                    RealmInstant::class -> assertEquals(
+                        expected.timestampField,
+                        decoded.timestampField
+                    )
+                    ObjectId::class -> assertEquals(expected.objectIdField, decoded.objectIdField)
+                    RealmUUID::class -> assertEquals(expected.uuidField, decoded.uuidField)
+                    MutableRealmInt::class -> assertEquals(
+                        expected.mutableRealmIntField,
+                        decoded.mutableRealmIntField
+                    )
+                    RealmObject::class -> {
+                        assertEquals(
+                            expected.nullableObject!!.stringField,
+                            expected.nullableObject!!.stringField
+                        )
+                        assertEquals(
+                            expected.realmEmbeddedObject!!.name,
+                            expected.realmEmbeddedObject!!.name
+                        )
+                    }
+                    else -> throw IllegalStateException("Untested type $classifier")
+                }
+            }
+    }
+
+    @Test
+    fun exhaustiveRealmAnyTester() {
+        val ignoredClassifiers = arrayOf(
```

Comment:
Then maybe just iterate `TypeDescriptor.anyClassiers` instead of this below:
```
TypeDescriptor
            .elementClassifiers
            .filterNot { ignoredClassifiers.contains(it) }
```